code/game/objects/items/robot/robot_upgrades.dm

```diff
@@ -211,3 +210,70 @@
 
 	R.SetEmagged(1)
 	return 1
+
+/obj/item/borg/upgrade/selfrepair
+	name = "self-repair module"
+	desc = "This module will repair the cyborg when it is in a cyborg recharger."
+	icon_state = "cyborg_upgrade5"
+	require_module = 1
+	var/repair_amount = -1
+	var/repair_cooldown = 0
+	var/msg_cooldown = 0
+	var/on = 0
+	var/powercost = 10
+	var/mob/living/silicon/robot/cyborg
+
+/obj/item/borg/upgrade/selfrepair/action(mob/living/silicon/robot/R)
+	if(..())
+		return 0
+	var/obj/item/borg/upgrade/selfrepair/U = locate() in R
+	if(U)
+		usr << "<span class='warning'>This unit is already equipped with a self-repair module.</span>"
+		return 0
+	cyborg = R
+	icon_state = "selfrepair_off"
+	action_button_name = "Toggle Self-Repair"
+	R << "<span class='notice'>Self-repair module detected. You will be repaired when inside an active recharge station.</span>"

```

Comment:
this feedback is misleading.
All the device does is take power form the cell, and then repair the cyborg. this is only tangentially related to recharges because rechargers recharge cell power.